### DISCUSSION TITLE
Blackframe, freeze and mute detection ?

### DISCUSSION THREAD

**Discussion by SebastianManemannEyez** [OvenMediaLabs/OvenMediaEngine#1470] [Ideas]
Are there any plans to add blackframe, freeze and mute detection as alarms ? Could be useful.

**Comment by nums** [comment]
This is not possible, I think this is not the scope of OvenMedia, however it would be interesting to be able to easily use a tool like openCV which allows detection / processing on frames:
- one possibility would be to be able to read an output stream in RTMP or SRT format but this is not implemented: https://github.com/AirenSoft/OvenMediaEngine/discussions/1408
- But the ideal would be to integrate a sort of "frame" hook which would allow you to retrieve a "raw" frame and carry out the processing / detection on openCV (or other) and in some case send the processed frame back to OvenMedia
